### PR TITLE
Computer wirepanel

### PR DESCRIPTION
**Pull request by Errant-4**
## About the PR
Computers now have basic wire"hacking" implemented. Their maintenance panel opens and closes with a screwdriver, like any other device with wires. They either have 1 Power Wire, 1 AI Wire and 2 Dummy Wires, or 1 Power Wire and 3 Dummy Wires if the computer does not have the AI connection whitelist.

Deconstruction steps changed from "Screw, Pry, Cut, Screw" to "Pry, Pry, Cut, Screw" with small doafters added for the first two Pries, so I don't think it's necessarily faster now, even though you don't need to change tools between the first two steps

~~Minor Sprite issue: there is no visual indication for the maintenance panel being open (but examine shows the current state)~~

~~Minor UI issue with deconstrction: after Prying once with the crowbar, the Screwdriver interaction will both try to open the service panel and finalise the deconstruction step. As such, there will be two doafter progress bars and the maintenance panel can be heard opening. nevertheless, once the slower progress bar finishes, the entity state will go to the next node and so the maintenance panel will "disappear"', until the machine is reconstructed. (by applying Screwdriver again, as normal)~~

## Why / Balance
This allows the crew to sever the AI from specific computers. It also makes it possible to disable the computer via the power wire, although there is not much point to this. Allows future maintenance panel or hacking-related features to be added to computers. 

## Technical details
Computers now have the required wirehacking components, and one of two wire configurations, depending on whether or not they have StationAiWhitelistedComponent.

Tests have been changed to fit the new deconstruction process

## Breaking changes
Computers consoles that are AI accessible now inherit from BaseComputerAiAccess.
Computer prototypes that redefine their sprites instead of inheriting the defaults must now also specify the sprite map/state for the maintenance panel.
Computers that define another UserInterface need to make sure to add WiresBoundUserInterface again so it is not overwritten. See BaseComputer for an example.

## Media

New "panel open" sprite:

image

https://github.com/user-attachments/assets/7873290b-db8d-47ec-9b7d-711ca05c9009

(obsolete)
https://github.com/user-attachments/assets/f39e2252-4d5a-463c-9257-ba6d57140471

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

**Changelog**
:cl: Errant
- tweak: Computers now have a maintenance panel with a Power and AI wire (if AI-connected). 
- tweak: Computer deconstruction now begins with the crowbar, not the screwdriver.

### CODE REVIEW

**Comment by Errant-4** [comment]
> the hero

You know what I was ALSO thinking, while making this? Giving them a second power wire, and when you snip one... B)

**Comment by ScarKy0** [comment]
> You know what I was ALSO thinking, while making this? Giving them a second power wire, and when you snip one... B)

You should go back to this thought as it would be funny

**Comment by Errant-4** [comment]
A... It doesn't like the simultaneous interactions during deconstruction, I think.

**Comment by baddiePRO** [comment]
ACC wire for comms console/crim record/etc?

**Comment by ScarKy0** [comment]
> ACC wire for comms console/crim record/etc?

This, an ACC wire would be great

**Comment by Errant-4** [comment]
I will lead all of the more controversial/debatable wire changes to separate PRs

**Comment by dvir001** [comment]
ACC wire is weird, have you ever seen a computer that just let you open it based on a cut wire?

File: Resources/Prototypes/Entities/Structures/Machines/Computers/computers.yml [hunk 1]
```diff
@@ -41,6 +41,11 @@
     interfaces:
         enum.AtmosAlertsComputerUiKey.Key:
             type: AtmosAlertsComputerBoundUserInterface
+        enum.WiresUiKey.Key:
+          type: WiresBoundUserInterface
+  - type: Wires
+    boardName: wires-board-name-computer
+    layoutId: ComputerAi
```
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Structures/Machines/Computers/computers.yml:48]: Make a new parent like this
```
- type: entity
  parent: BaseComputer
  id: BaseComputerAiAccess
  components:
  - type: StationAiWhitelist
  - type: Wires
    boardName: wires-board-name-computer
    layoutId: ComputerAi
```
so you don't have to copy paste that part everywhere

**Comment by slarticodefast** [comment]
There should probably also be a visual indication for an open panel like there is for other machines.

File: Resources/Prototypes/Entities/Structures/Machines/Computers/computers.yml (hunk 1 above)
Comment (Errant-4) [reply Resources/Prototypes/Entities/Structures/Machines/Computers/computers.yml]: Makes sense, thanks

**Comment by ItsMeThom** [comment]
Something like an access wire (like say, snipping it allows non cargo-techs to order cargo) is probably a separate PR after this right?

**Comment by superjj18** [comment]
> ACC wire is weird, have you ever seen a computer that just let you open it based on a cut wire?

The NES CIC drm chip

https://youtube.com/shorts/EP7F7g1ZC1Q?si=B_PAGuxwKIb0pbe7

**Review by slarticodefast — APPROVED** [review]
Tested in-game, everything works as intended.

**Comment by Errant-4** [comment]
I have added Breaking Change notes for downstreams on how to update any custom computers

**Pull request merged**